### PR TITLE
refactor out uchiwa::params to init.pp, add new users array option

### PR DESCRIPTION
**Pull request by countryHick**
to help (1st cut with) #46 ...

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -4,11 +4,22 @@
...
 class uchiwa::config inherits uchiwa {
 
-  file { '/etc/sensu/uchiwa.json':
-    ensure  => file,
-    content => template('uchiwa/etc/sensu/uchiwa.json.erb'),
-    owner   => uchiwa,
-    group   => uchiwa,
-    mode    => '0440',
+
+  if count($uchiwa::users) > 0 {
```
Comment (solarkennedy) [inline manifests/config.pp]: Please do not have two entrypoints for when users or set or not.
Reduce duplication here, just have one template

File: manifests/init.pp [hunk 2]
```diff
@@ -96,23 +96,34 @@
...
-  $manage_services      = $uchiwa::params::manage_services,
-  $manage_user          = $uchiwa::params::manage_user,
-  $host                 = $uchiwa::params::host,
-  $port                 = $uchiwa::params::port,
-  $user                 = $uchiwa::params::user,
-  $pass                 = $uchiwa::params::pass,
-  $refresh              = $uchiwa::params::refresh,
-  $sensu_api_endpoints  = $uchiwa::params::sensu_api_endpoints,
-) inherits uchiwa::params {
+  $package_name         = 'uchiwa',
```
Comment (solarkennedy) [inline manifests/init.pp]: I agree that we were not really using params at all.

File: templates/etc/sensu/users-uchiwa.json.erb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      "path": "<%= endpoint['path'] || default_endpoint['path'] %>",
+      "timeout": <%= endpoint['timeout'] || default_endpoint['timeout'] %>
+    }<%= ',' if i < (@sensu_api_endpoints.size - 1) %>
+    <%- end -%>
+  ],
+  "uchiwa": {
+    "host": "<%= @host %>",
+    "port": <%= @port %>,
+    "users": [
+      <%- @users.each_with_index do |user, i| -%>
```
Comment (solarkennedy) [inline templates/etc/sensu/users-uchiwa.json.erb]: This needs some sort of unit test to prove that it works

File: manifests/init.pp [hunk 4]
```diff
@@ -96,23 +96,34 @@
...
+  $repo_source          = undef,
+  $repo_key_id          = '7580C77F',
+  $repo_key_source      = 'http://repos.sensuapp.org/apt/pubkey.gpg',
+  $manage_services      = true,
+  $manage_user          = true,
+  $host                 = '0.0.0.0',
+  $port                 = '3000',
+  $user                 = '',
+  $pass                 = '',
+  $users                = [],
```
Comment (solarkennedy) [inline manifests/init.pp]: This needs a docstring on the format

**Comment by solarkennedy** [comment]
Unit tests need to be addressed.

**Comment by countryHick** [comment]
closing for now - need rethink :(

File: manifests/init.pp (hunk 4 above)
Comment (countryHick) [inline manifests/init.pp]: what do mean by docstring -- as in puppet docs?

Comment (solarkennedy) [inline manifests/init.pp]: Yea... but looking at this now I don't know what it is in reference to.

Comment (countryHick) [inline manifests/init.pp]: i'm not sure i understand what you mean - but it is in line with the
puppet-sensu module, and it's got the docs above as well regardless...

**Pull request closed without merging**